rafx-api/src/backends/gles3/command_buffer.rs

```diff
@@ -0,0 +1,1168 @@
+                        }
+                        RafxSamplerIndexGles3::Mutable(sampler_descriptor_index) => {
+                            // Find the descriptor with the relevant sampler
+                            let sampler_descriptor =
+                                root_signature.descriptor(sampler_descriptor_index).unwrap();
+                            // Find the samplers within the descriptor set's flattened array of
+                            // all samplers
+                            let sampler_index = (array_index * data.sampler_states_per_set
+                                + sampler_descriptor.descriptor_data_offset_in_set.unwrap())
+                                as usize;
+                            &data.sampler_states[sampler_index].as_ref().unwrap().sampler
+                        }
+                    };
+
+                    let layout = &root_signature.inner.layouts[set_index as usize];
+
+                    for i in 0..descriptor.element_count {
+                        if let Some(location) =
+                            pipeline_info.resource_location(descriptor.descriptor_index, i)
+                        {
+                            let image_state_index = base_image_state_index + i;
+                            let texture = &data.texture_states[image_state_index as usize]
+                                .as_ref()
+                                .expect("Tried to use unbound texture")
+                                .texture;
+
+                            let texture_unit_index: u32 = layout.texture_unit_offset
+                                + descriptor.descriptor_data_offset_in_set.unwrap()
+                                + i;
+
+                            //println!("texture unit {} for location {:?}", texture_unit_index, location);
+                            gl_context.gl_active_texture(texture_unit_index)?;
+
+                            let target = texture.gl_target();
+
+                            //TODO: Handle specific mip levels/array slices (GL_TEXTURE_BASE_LEVEL and GL_TEXTURE_MAX_LEVEL on sampler, ES3 only)
+
+                            gl_context.gl_bind_texture(
+                                target,
+                                texture.gl_raw_image().gl_texture_id().unwrap(),
+                            )?;
+
+                            gl_type_util::set_uniform(
+                                gl_context,
+                                location,
+                                &texture_unit_index,
+                                gles3_bindings::INT,
+                                1,
+                            )?;
+
+                            let min_filter = if texture.texture_def().mip_count > 1 {
+                                sampler.inner.gl_mip_map_mode
+                            } else {
+                                sampler.inner.gl_min_filter
+                            };
+
+                            gl_context.gl_tex_parameteri(
```

Comment:
Should use sampler objects, only available in ES3

Reply:
Also we may need to support specific array layer/mip levels or other texture types